networkx/algorithms/cluster.py

```diff
@@ -102,6 +103,14 @@ def _weighted_triangles_and_degree_iter(G, nodes=None, weight="weight"):
     def wt(u, v):
         return G[u][v].get(weight, 1) / max_weight
 
+    def cubed_root(num):
+        if num > 0:
+            return pow(num, 1.0 / 3.0)
+        elif num < 0:
+            return -pow(abs(num), 1.0 / 3.0)
+        else:
+            return 0
```

Comment:
```suggestion
        if num >= 0:
            return pow(num, 1 / 3)
        return -pow(-num, 1 / 3)
```
I suppose you could use `numpy.cbrt` also...

Reply:
From what I could gather, sign-adapted Zhang performs better than sign-adapted Onnela.

They made a few observations:
1. The coefficient that adapted Zhang produces is more **linear** than the other 2 algorithms ( when produced as a function of the proportion of negative triangles ), which makes it more human-understandable.
2. Adapted Zhang and adapted Onnela are more **noise-resistant** than adapted Watts, with similar performance when the threshold for noise is raised.

I will revert the changes for now, and work on directed graphs some other time.

Do you have any suggestions for how to proceed in implementing adapted Zhang? 
Should a signed, weighted graph be explicitly specified by the user?

Thank you for the help and direction!

Reply:
From what I could understand, adapted Zhang is less impacted than adapted Onnela by random noise.

I will revert the changes and leave clustering for directed graphs for some other time.

Do you have any suggestions on how to proceed from here?
Thank you for the help and direction.